### ISSUE TITLE
Log what resources are created in K8s

### ISSUE THREAD

**Issue by jsm84** [operator-framework/helm-app-operator-kit#52]
I ran into an issue debugging the example Tomcat operator (and a MariaDB example of my own), where the operator processes `cr.yaml` and creates the resources transparently (K8s Deployment and Service objects defined by the chart), but pods themselves are never created as this action was blocked by an OpenShift Security Context Constraint. It appeared as though the operator wasn't actually deploying resources when processing `cr.yaml`, however it actually did, it just wasn't obvious by viewing the output of `oc logs <operator-pod-id>`.

After giving the code a rough glance, it appears that these messages are logged via logrus. It would be ideal to have something logged in addition to `processing <namespace>/my-tomcat using values: replicaCount: 2` which states the Kind and the name of resources that are created by the Helm operator.

**Comment by robszumski** [comment]
Seems to be a few concerns here:
 - do we rely on logging directly from Helm or can we customize it?
   - showing a git style diff in the logging output is pretty helpful
 - do we use log levels today?
   - if so, how can you control it?
 - does the Operator write out to the status block?

**Comment by jsm84** [comment]
To quickly answer whether the "status" field gets written to, it does get populated with info that differs somewhat from what is logged to the pods stdout (which is what I believe populates the output of `oc logs <operator-pod-id>`).

Looking at the `status:` block of the tomcat-operator pod shows the following:
```
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: 2018-10-13T01:10:50Z
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: 2018-10-15T18:23:07Z
    status: "True"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: 2018-10-13T01:10:49Z
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://081324cc0935ed07b299e0b964beb175d0d796e07494985f8b167409b8734b53
    image: scan.connect.redhat.com/p7153749677c0f9fde6d2669c3d7785cffc2124c4f/tomcat-operator:0.1.5
    imageID: docker-pullable://scan.connect.redhat.com/p7153749677c0f9fde6d2669c3d7785cffc2124c4f/tomcat-operator@sha256:bf76931832fa3559029d54e8f0908b0d108da9453c96ed34a2de26b662984ac6
    lastState:
      terminated:
        containerID: docker://a6f1f4cce926976505c914d4aac8a8398e129c4a7327c6037dbdcaa956d2ac48
        exitCode: 1
        finishedAt: 2018-10-15T18:20:44Z
        reason: Error
        startedAt: 2018-10-15T18:20:40Z
    name: tomcat-operator
    ready: true
    restartCount: 3
    state:
      running:
        startedAt: 2018-10-15T18:23:06Z
  hostIP: 192.168.124.122
  phase: Running
  podIP: 172.17.0.9
  qosClass: BestEffort
  startTime: 2018-10-13T01:10:50Z
```

Whereas the (truncated) log output of the operator is:
```
time="2018-10-15T18:23:07Z" level=info msg="Go Version: go1.10.4"
time="2018-10-15T18:23:07Z" level=info msg="Go OS/Arch: linux/amd64"
time="2018-10-15T18:23:07Z" level=info msg="operator-sdk Version: 0.0.6+git"
time="2018-10-15T18:23:08Z" level=info msg="Registering Components."
time="2018-10-15T18:23:09Z" level=info msg="Watching apache.org/v1alpha1, Tomcat, tomcat-operator, 5000000000"
time="2018-10-15T18:23:09Z" level=info msg="Starting the Cmd."
time="2018-10-15T18:23:09Z" level=info msg="processing tomcat-operator/my-tomcat"
time="2018-10-15T18:23:09Z" level=info msg="using values: replicaCount: 2\n"
time="2018-10-15T18:23:09Z" level=error msg="a release named helm-app-operator-my-tomcat already exists.\nRun: helm ls --all helm-app-operator-my-tomcat; to check the status of the release\nOr run: helm del --purge helm-app-operator-my-tomcat; to delete it"
time="2018-10-15T18:23:10Z" level=info msg="processing tomcat-operator/my-tomcat"
time="2018-10-15T18:23:10Z" level=info msg="using values: replicaCount: 2\n"
time="2018-10-15T18:23:10Z" level=info msg="adding ownerrefs to file: tomcat/templates/appsrv-svc.yaml"
time="2018-10-15T18:23:10Z" level=info msg="adding ownerrefs to file: tomcat/templates/appsrv.yaml"
time="2018-10-15T18:23:23Z" level=info msg="processing tomcat-operator/my-tomcat"
time="2018-10-15T18:23:23Z" level=info msg="skipping tomcat-operator/my-tomcat because resource version has not changed"
time="2018-10-15T18:23:28Z" level=info msg="processing tomcat-operator/my-tomcat"
time="2018-10-15T18:23:28Z" level=info msg="skipping tomcat-operator/my-tomcat because resource version has not changed"
time="2018-10-15T18:23:33Z" level=info msg="processing tomcat-operator/my-tomcat"
time="2018-10-15T18:23:33Z" level=info msg="skipping tomcat-operator/my-tomcat because resource version has not changed"
...
```

**Comment by joelanford** [comment]
Great timing @jsm84. I don't think this will completely answer your need, but check out the latest PR I submitted this morning (#51), which includes a commit with a bunch of logging updates that makes it more obvious what's happening (and feel free to leave comments).

A few things not present in that PR that still need to be done are:
* Configurable logging level. Right now there's no way to see debug logs.
* Logging a git style diff when changes made to the CR trigger a release installation or update.

> Looking at the status: block of the tomcat-operator pod...

I think the status @robszumski was referring to is the CR status, not the helm operator. It contains the full release information (e.g. release name and manifests) for the currently deployed release.

**Comment by robszumski** [comment]
Whoa that's a hell of a status ;)

**Comment by joelanford** [comment]
Closed by #61

We'll track log level configurability in #63.
